DCAT-2416 Refactor navigation and editorial updates

meker12 opened the pull request:
## Purpose of this pull request

This PR refactors the top navigation for Merchandising Services and creates stand-alone API reference pages for the Data Ingestion REST API, and the Merchandising Services GraphQL API

- Move authentication for Data ingestion API to it's own topic
  - Clarify that this auth is for direct, interactive access, and link to User Auth Implementation guide for more info 
 - Change "Using topics" to getting started
- Add API Reference header
- Create a reference folder for stand-alone API reference pages
- Update spectaql storefront metadata to remove CategoryView and CategoryViewInterface from the Merchandising Services API reference
- Fix landing page to match what's on developer.adobe.com
- Add reference folder with API reference pages 
- Editorial changes to remove references to the ACO Admin API and redirect to ACO UI.
- Add edition tags

meker12 commented on src/pages/composable-catalog/ccdm-use-case.md:
```diff
@@ -11,7 +11,7 @@ keywords:
 # Using the composable catalog with your storefront
 
-Learn how a company with a single base catalog can use the Merchandising Services powered by Channels and Policies APIs to add product data, define catalogs using projections, and retrieve the catalog data for display in a headless storefront.
+Learn how a company with a single base catalog can use the Merchandising Services powered by Channels and Policies APIs to ingest catalog data, and retrieve catalog data for display in a headless storefront.
```
```suggestion
Learn how a company with a single base catalog can use the Merchandising Services powered by Channels and Policies APIs to ingest catalog data into Adobe Commerce Optimizer and retrieve it for display in a headless storefront.
```

jeff-matthews reviewed (commented):
Two things stand out on cursory review:

- The links to API Reference 404. Is that expected?
- The `edition: ee` metadata key-value pair is no longer supported, which is why pages with that metadata are rendering the fallback yellow badge. You'll need to update that.

meker12 commented:
> Two things stand out on cursory review:
> 
> * The links to API Reference 404. Is that expected?
> * The `edition: ee` metadata key-value pair is no longer supported, which is why pages with that metadata are rendering the fallback yellow badge. You'll need to update that.

I fixed the 404 links. I can't test the API reference correctly until we push to the staging branch -- tried building on gitpages pipeline, but the build fails.

Added edition tags.

jeff-matthews reviewed (approved):
LGTM 😄
